andela/Ah-backend-xmen
# #162725261 Implement commenting on articles

#### What does this PR do?
Enable commenting on articles

#### Description of Task to be completed?
- create a comment model class to handle CRUD operations on Comments
- create a serailzer class for the model Comments
- create views to handle all CRUD operations for Comments Model
- add routes for the comment views
 
#### How should this be manually tested?
- Fetch and check out this branch with `git fetch origin ft-comment-on-articles-162725261 and git checkout ft-comment-on-articles-162725261` 
- Create and activate your virtual environment with `python3 -m virtualenv <preferredname>` and `source <preferredname>/bin/activate`
- run migrations with `python manage.py migrate comments`
- Start the app by running `python manage.py runserver`
- To comment on an article perform a post on `localhost:8000/api/articles/<slug>/comments` passing json in data `{"body":"body data"}` **requires authentication**
- To view all comments perform a get method on `localhost:8000/api/articles/<slug>/comments`
- To view a single article perform a get request on `localhost:8000/api/articles/<slug>/comments/pk/`
- To update article comment peform a put on `localhost:8000/api/articles/<slug>/comments/pk/` to **be done by person who created the comment**
- To delete article comment perform a delete on `localhost:8000/api/articles/<slug>/comment/pk/` to **be done by person who created the comment**
- To reply to a comment perform a `POST` request on end point `localhost:8000/api/articles/<slug>/comment/pk/reply/`
- To view all replies on a comment perform a `GET` on `localhost:8000/api/articles/<slug>/comment/pk/reply/`
- To view,update,delete a specific reply of a user pass in reply_id in `localhost:8000/api/articles/<slug>/comment/pk/reply/<reply_id>/` **but to be able to delete or update a reply you must be the author of the reply**
- To test this branch run locally run `python manage.py test authors/apps/comments/`


#### What are the relevant pivotal tracker stories?
[ #162725261](https://www.pivotaltracker.com/story/show/162725261)


[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/fd09e0fef2dd4b779a00)

**Screenshots**
<img width="1406" alt="screen shot 2019-01-22 at 5 11 39 pm" src="https://user-images.githubusercontent.com/39124174/51541173-3a4c7780-1e69-11e9-9243-776b824f4a9e.png">
<img width="1425" alt="screen shot 2019-01-22 at 5 12 20 pm" src="https://user-images.githubusercontent.com/39124174/51541174-3a4c7780-1e69-11e9-9333-f09bc55ea7e3.png">
<img width="1398" alt="screen shot 2019-01-22 at 5 12 41 pm" src="https://user-images.githubusercontent.com/39124174/51541175-3ae50e00-1e69-11e9-9e15-d8fa348e9ddc.png">
<img width="1437" alt="screen shot 2019-01-22 at 5 12 57 pm" src="https://user-images.githubusercontent.com/39124174/51541176-3ae50e00-1e69-11e9-9d9c-43556133e240.png">
<img width="1433" alt="screen shot 2019-01-22 at 5 13 10 pm" src="https://user-images.githubusercontent.com/39124174/51541177-3b7da480-1e69-11e9-80eb-45367139aa31.png">
![screen shot 2019-01-24 at 9 45 59 am](https://user-images.githubusercontent.com/39124174/51676666-62161980-1fe8-11e9-8bc8-f3c0ffeffd74.png)
![screen shot 2019-01-24 at 9 46 44 am](https://user-images.githubusercontent.com/39124174/51676667-62161980-1fe8-11e9-916f-ee55108eb615.png)
![screen shot 2019-01-24 at 10 06 34 am](https://user-images.githubusercontent.com/39124174/51676669-62aeb000-1fe8-11e9-97c3-760324458c8e.png)
![screen shot 2019-01-24 at 10 09 38 am](https://user-images.githubusercontent.com/39124174/51676670-62aeb000-1fe8-11e9-8bec-1a33eae262be.png)
![screen shot 2019-01-24 at 10 14 30 am](https://user-images.githubusercontent.com/39124174/51676671-63474680-1fe8-11e9-9f05-e4112e565ad5.png)
![screen shot 2019-01-24 at 10 14 41 am](https://user-images.githubusercontent.com/39124174/51676672-63474680-1fe8-11e9-8159-8d34cf5c4096.png)
